### PR TITLE
Rewrite the Tekton Pipelines overview for clarity and flow

### PR DESCRIPTION
**Pull request by spomorski**
# Changes

Rewrites the Tekton Pipelines overview for clarity and reorganizes the information for better flow.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,67 +1,63 @@
...
-
-## Related info
-
-If you are interested in contributing to the Tekton Pipeline project, see the
-[Tekton Pipeline contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md).
+Tekton Pipelines is a custom Kubernetes extension that installs and runs on your Kubernetes cluster.
+It defines a set of Kubernetes [Custom Resources](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) that act as building blocks from which you can assemble custom CI/CD pipelines. Once installed,
+Tekton Pipelines becomes available via the Kubernetes CLI (kubectl) and via API calls, just
+like pods and other resources. Tekton is open-source and part of the [CD Foundation](https://cd.foundation/),
+a [Linux Foundation](https://www.linuxfoundation.org/projects/) project.
```
Comment (sbwsg) [inline docs/README.md]: We use the word "custom" several times in this paragraph but it doesn't add a ton of meaning except when it's the name of a specific Kubernetes type (Custom Resource).  Suggest dropping its usage.

File: docs/README.md [hunk 2]
```diff
@@ -1,67 +1,63 @@
...
+
+## Tekton Pipelines entities
+
+Tekton Pipelines defines the following entities:
+
+<table>
```
Comment (sbwsg) [inline docs/README.md]: Github supports formatting tables without HTML ([see their help docs here](https://help.github.com/en/github/writing-on-github/organizing-information-with-tables)) which is a little more readable than the raw HTML I think.

File: docs/README.md (hunk 1 above)
Comment (spomorski) [reply docs/README.md]: Agreed, I was being over-specific as a newbie to the space. Removed.

File: docs/README.md (hunk 2 above)
Comment (spomorski) [reply docs/README.md]: I'm aware of markdown tables, but in my experience I've found them cumbersome and always had better visual results with HTML when a table was more complex than 2-3 columns. If we prefer markdown tables in our docs as a rule, though, I'll convert it, of course.

Comment (sbwsg) [reply docs/README.md]: I have no strong feeling either way, just wanted to see if you knew of them already.  If your prefer authoring with the html then that's cool by me! :+1:

Comment (spomorski) [reply docs/README.md]: Cool, I'll stick with HTML then. :)

**Pull request merged**